client-server-client/src/main/scala/zio/raft/client/ActionStream.scala

```diff
@@ -0,0 +1,341 @@
+package zio.raft.client
+
+import zio._
+import zio.stream._
+import zio.raft.protocol._
+
+/**
+ * Client-side unified action stream processing.
+ * 
+ * Merges multiple event sources into a unified stream:
+ * - Network messages from server (ZeroMQ)
+ * - User client requests  
+ * - Timer events (timeouts, keep-alives)
+ * 
+ * Produces connection state-aware actions and maintains
+ * a separate stream for server-initiated requests.
+ */
+trait ActionStream {
```

Comment:
this is too verbose, we don't need this class at all.

All of it should be part of the RaftClient, and we don't need a trait. Keep it simple. Also, we don't need a test and impl. Just part of RaftClient